.github/workflows/build.yaml

```diff
@@ -323,6 +323,27 @@ jobs:
                   path: |
                       .environment/gn_out/.ninja_log
                       .environment/pigweed-venv/*.log
+            - name: Delete Defaults
+              run: defaults delete com.apple.dt.xctest.tool
+              continue-on-error: true
+            - name: Run macOS Build
+              timeout-minutes: 40
+              # Enable -Werror by hand here, because the Xcode config can't
+              # enable it for various reasons.  Keep whatever Xcode settings
+              # for OTHER_CFLAGS exist by using ${inherited}.
+              #
+              # Disable -Wmacro-redefined because CHIP_DEVICE_CONFIG_ENABLE_MDNS
+              # seems to be unconditionally defined in CHIPDeviceBuildConfig.h,
+              # which is apparently being included after CHIPDeviceConfig.h.
+              run: xcodebuild -target "Matter" -sdk macosx OTHER_CFLAGS='${inherited} -Werror -Wno-macro-redefined'
```

Comment:
I'll triple check, but the apps build below failed without it.... at least I thought they did.

Reply:
Can confirm, this does require building Matter.framework ahead of time:
https://github.com/project-chip/connectedhomeip/runs/7143474926?check_suite_focus=true

Reply:
OK.  I guess that's because darwin-framework-tool is in the "standalone" builds in BUILD.gn.

But maybe it would be better to just turn off building darwin-framework-tool here instead?  There's not much point to it, since we build it as part of the Darwin Tests job _and_ part of the Darwin job....

Reply:
Concretely, @woody-apple add `enable_standalone_chip_tool_darwin_build=false` to the `GN_ARGS` where we are doing `target_os="all"`.

Reply:
Done, thanks!